### ISSUE TITLE
Restoring WW problems

### ISSUE THREAD

**Issue by bnmnetp** [RunestoneInteractive/rs#373]
Alex,

Moving this to an issue... so we can track more precisely than email.  Tagging @Tanaquil18 and @mattboelkins

This morning I was able to implement some changes to my side of the webwork code that filled in some deficiencies. The result of which is:

1. no more toggle behavior.  The icon reliably appears and *most* answers are reliably restored.
2. The server is updated with the correct percent correct when the student submits.  This is not new, just confirming that the server side autograder has the right data to do its job correctly.
3. Following longstanding Runestone behavior, the latest student answer is stored in local storage on the browser.  When the student brings up a page, we get the answer from the server.  If the answer from the server indicates that it is correct then we accept the answer from the server.  If the answer from the server is not correct then we check local storage.  If the answer from local storage is correct the server is updated.  If the answer from local storage is *newer* than the server then we update the server.
4. Fill in the blank answers always repopulate correctly.
5. drop down menu answers seem flakey - sometimes it seems they are just lost.  I may not be giving Alex enough data or the right data to restore these....
6. Multiple choice answers behave similarly.  again I may not be giving alex what he needs.

Alex, here is what you have to work with....  Seems likely something is missing.

<img width="556" alt="Screenshot 2023-09-23 at 9 38 57 AM" src="https://github.com/RunestoneInteractive/rs/assets/51115/6db2ac42-06cf-479c-828a-2857355fd755">

**Comment by Alex-Jordan** [comment]
Thanks, I will continue with this with the original problem that Chrissy pointed to and see what more I can either do, or at least diagnose.

**Comment by bnmnetp** [comment]
@Alex-Jordan Anything to report on this one?

**Comment by Alex-Jordan** [comment]
No, but thanks for the reminder. I'll try to look today. Do you recall if this data was from the same exercise Chrissy reported with before?

**Comment by bnmnetp** [comment]
Yes, it was.

**Comment by Alex-Jordan** [comment]
Something is screwy with the WW problem in that example, and it is screwy in a way that overlaps with this issue. So without getting into the details here, I want to get to the bottom of that screwiness before I would continue. I'm 95% certain that the problem has an issue with its PG code, or it is exposing a bug with a PG macro library that it relies on.

But this copy of AC is using a WW host course on Sean's server, and this problem comes from a PG problem file that I believe Chrissy wrote and put in that course on Sean's server. That means I don't have access to look at the code for that problem. Also I trust Sean's server is up to date with hotfixes and does not have troublesome customizations, but I can't directly confirm that. So these are significant issues with using Sean's server for a Runestone Academy-hosted book.

@Tanaquil18 and @mattboelkins, are there instances of this issue (with the radio buttons, checkboxes, or dropdown selects not repopulating) that you could direct me to, where the WW problem is either coded in PTX source of the book, or it is an OPL problem?

**Comment by Tanaquil18** [comment]
Totally possible I'm doing something wrong in writing problems... I was doing my best to rewrite things so that we'd have a functioning copy of AC, and if the problems worked on the WeBWorK server and produced valid PTX then I called it good, but I don't know what I don't know. 

We know that it wasn't ideal to keep using Sean's server, but we were up against a time crunch with our semesters starting, and the default AIM course didn't work, and then we didn't want to mess with what was working while instructors were using it with students. My hope/plan is that during the semester break we can switch to the webwork.runestone.academy server you set up.    

I think you should have access to examine my code though, because you loaded my folder of problems in both the default aim course and the webwork.runestone.academy course.  But that likely won't help you with testing since we're not actually using either of those.  I am trying to work on adding these kinds of problems to a short book for Runestone testing.

The only problems hardcoded in the source are the Chapter 8a Previews that you wrote, and those all do seem to restore answers to multiple choice and dropdowns.  Not sure if that's because they're hardcoded, or because they were written by an expert. :-)

But the first one I found that came from the OPL and has multiple choice doesn't restore answers either.  [Exercise 1](https://runestone.academy/ns/books/published/franciscan_mth162_F23/sec-8a-3-geometric.html#Library__Michigan__Chap9Sec2__Q05) 
https://runestone.academy/ns/books/published/franciscan_mth162_F23/sec-8a-3-geometric.html#Library__Michigan__Chap9Sec2__Q05 which comes from

```
<exercise label="Library__Michigan__Chap9Sec2__Q05">
    <webwork source="Library/Michigan/Chap9Sec2/Q05.pg"/>
  </exercise>
```
in the source.

A lot of the multiple choice, checkboxes, and dropdowns that appear in ac-single were edited by me, to replace essay boxes, and/or to update the old checkbox macro (which doesn't say right/wrong in Runestone when answers are submitted) to the newer parserCheckboxList, and/or to rewrite the radio buttons in the newer style with double braces so that the answers will randomize, instead of the older style with single braces where the answers are given in the order they were coded in.  I didn't get to all of them, but it does mean that most of the problems left from the OPL are old-style.  Most of my edited ones are new style, but could easily have something else screwy.

**Comment by Alex-Jordan** [comment]
Thanks, I'll look at that OPL problem in context of AC on Runestone when I can, maybe today.

If you want to pursue the "screwiness" with that other one, here i what part of its static form looks like to me:

<img width="585" alt="Screenshot 2023-10-07 at 8 08 23 PM" src="https://github.com/RunestoneInteractive/rs/assets/4732672/8547b070-8aee-475c-b785-36042a59b2bf">

And here is that same part after I activate it:

<img width="545" alt="Screenshot 2023-10-07 at 8 08 51 PM" src="https://github.com/RunestoneInteractive/rs/assets/4732672/39e7ab97-ddb6-43d8-9fb6-2f4842c0c25a">

Not sure what's up there, but I can't trust answers will repopulate right if I can't trust the answers even stay the same from instance to instance. If you don't see anything in your code to explain this, it could be a bug with radioButtons and I'll have to follow up later.

**Comment by Tanaquil18** [comment]
Oh, I see the issue with that.  I added the double brackets so that the options would randomize, but I didn't correctly tell it to keep the labels with their original choices.  I noticed that mistake at some point when I was editing problems, but it must have been after this problem and I didn't go back and fix this one.  I just fixed it, but it won't be visible until we rebuild with webwork generation.  

That issue won't appear in the other one I just pointed out, or in the other one from the original post which had dropdowns and no labels, [Preview Activity 5.5.1](https://runestone.academy/ns/books/published/franciscan_mth162_F23/sec-5-5-other-opt.html#ACS__Preview__5-5) 
https://runestone.academy/ns/books/published/franciscan_mth162_F23/sec-5-5-other-opt.html#ACS__Preview__5-5

Thanks for pointing it out!

**Comment by Alex-Jordan** [comment]
OK, I think we have been playing whack-a-mole for a while. I now understand there are two types of radio buttons question. One type was given a "label" in the PG source (I put in quotes because it is not the HTML label). The "label" could be "A", "B", etc or "distance", "acceleration", etc. If there is a label, the HTML input for that radio button has value="..." whatever the label was. Also, this "label" is stored as the answer that a user submitted.

If the PG problem did _not_ assign labels, then the value for the HTML input is automatically created to be B0, B1, B2, etc. But this was obnoxious for traditional use of WeBWorK, and behind the scenes the actual answer that is used is "Choice 1", "Choice 2", "Choice 3". So Brad has stored something like "Choice 2" in these cases and I have to align that with "B1" in the new page where we are repopulating answers.

I think that it's been working for one type, then I get a report that it's not working somewhere, and I end up making it work there but unaware that I broke it the other place. But now I am putting code commentary in to not mess this up again going forward.

So in summary, see if you still find issues with radio button problems. And then next up, point me to either a select list (aka popup or dropdown) or to a checkboxes type question where we still have work to do.

**Comment by Tanaquil18** [comment]
The exercise from Chapter 8 I had linked before now works - its radio buttons restore answers!  I checked another problem with labels A, B, C, D, and that one works, too.  

But this one has radio buttons that still aren't restoring (and I had noticed the word label problem by this point in the summer, and my solution at the time was to make the options not randomize the order so that the labels would always be correct.) [Preview Activity 1.5.1](https://runestone.academy/ns/books/published/franciscan_mth162_F23/sec-1-5-units.html#ACS__Preview__1-5) 
https://runestone.academy/ns/books/published/franciscan_mth162_F23/sec-1-5-units.html#ACS__Preview__1-5

This one has both dropdowns and checkboxes.  The dropdowns in part a don't restore, and neither does at least the first checkbox in the second part (I didn't check the second one).  [Preview Activity 1.6.1](https://runestone.academy/ns/books/published/franciscan_mth162_F23/sec-1-6-second-d.html#ACS__Preview__1-6) 
https://runestone.academy/ns/books/published/franciscan_mth162_F23/sec-1-6-second-d.html#ACS__Preview__1-6

This one has only dropdowns, and none of them restore. [Preview Activity 5.5.1](https://runestone.academy/ns/books/published/franciscan_mth162_F23/sec-5-5-other-opt.html#ACS__Preview__5-5) 
https://runestone.academy/ns/books/published/franciscan_mth162_F23/sec-5-5-other-opt.html#ACS__Preview__5-5

**Comment by Alex-Jordan** [comment]
I looked at the PA 1.5.1 which is the next one that is still a radio buttons issue. (@Tanaquil18 , thanks for identifying those other problems, I'll look into them after this one is resolved.) For this one, I think that we need to change something in `rs/bases/rsptx/interactives/runestone/webwork/js/webwork.js`.

@bnmnetp, there are two places in that file that use `original_student_ans`, but I believe that should change to `student_value`. Actually I am pretty sure we need to make that change at line 101, but I am unsure about what `actString` is at line 104, and if the same change should be made there.

I do not know how you would like to go about making a change like this. Is it something to do in a test environment first?

Once that change is made, I will have a much easier time getting the radio button problems to all behave. Based on what I've seen so far, it will not hurt other exercises. For example in that problem there is one answer blank where the correct answer is like `1.19444 mi/min`. Right now using `original_student_ans`, I type in an answer (with only 4 sig figs) and it gets saved to the RS database as `1.194 (mi)/(min )`. With the change, my answer would be saved as `1.194 mi/min`. So should not cause harm.

But meanwhile with this kind of radio button problem, instead of saving something like `constant speeds`, it will save `B1` which is what I really need to identify the right button to mark as selected.

**Comment by bnmnetp** [comment]
Alex, yeah, this is exactly what the new sample chapter book will be useful for testing.

**Comment by Alex-Jordan** [comment]
I am about to add exercises that attempt to survey all radio button generating methods to that sample chapter. It would help avoid a merge conflict if you could comment on https://github.com/PreTeXtBook/pretext/pull/2086. It put a specific RS testing section into that doc.

**Comment by Alex-Jordan** [comment]
Earlier I suggested replacing `original_student_ans` with `student_value` at a certain place in the RS code. I take that back now (I have reasons, just don't want to distract by explaining here).

Instead I looked more closely at WW internals and I see how we can target more surgically to grab `student_value` (instead of `original_student_ans`) only when it is a radio buttons exercise. That will handle the last case that Chrissy has found of a radio buttons problem not having previous answers restored.

I will open a PR to `rs` for the change that is needed. However I do not yet understand something in this file: [`rs/bases/rsptx/interactives/runestone/webwork/js/webwork.js`](https://github.com/RunestoneInteractive/rs/blob/main/bases/rsptx/interactives/runestone/webwork/js/webwork.js). Note that `original_student_ans` appears twice. I understand its purpose at line 101. But what is the purpose of `actString` on line 104?

**Comment by Alex-Jordan** [comment]
OK, now the MC answers in the WW sample document are restoring previous answers. And also the three examples Chrissy listed here a few posts back from Active Calculus.

@Tanaquil18 Do you have time to confirm that these examples work now? And more time to look for any examples where things are not working? Today or tomorrow I have to pause any further edits to the JS for this, while other PTX people reorganize all our JS management.

**Comment by Tanaquil18** [comment]
I just tried checking with [Preview 1.1](https://runestone.academy/ns/books/published/franciscan_mth162_F23/sec-1-1-vel.html#ACS__Preview__1-1) and it's doing the same thing it used to with staying stuck on loading.

I tried with [Preview 1.5](https://runestone.academy/ns/books/published/franciscan_mth162_F23/sec-1-5-units.html#ACS__Preview__1-5) and the multiple choice button answers are not being restored after re-activating.

**Comment by Tanaquil18** [comment]
The Preview 1.1 console error is different than before though.

Uncaught TypeError: Cannot read properties of null (reading 'setAttribute')
    at Object.success (pretext-webwork.js:177:13)
    at u (jquery.min.js:2:27457)
    at Object.fireWith [as resolveWith] (jquery.min.js:2:28202)
    at k (jquery.min.js:2:77651)
    at XMLHttpRequest.<anonymous> (jquery.min.js:2:79907)

**Comment by Tanaquil18** [comment]
[Preview 5.5](https://runestone.academy/ns/books/published/franciscan_mth162_F23/sec-5-5-other-opt.html#ACS__Preview__5-5) is not restoring answers to the dropdowns, but that was no different than before.  However, a new behavior which is concerning is that it is changing the first answer to "None of these" after hitting submit.  

I entered u-sub for the first answer, and hit submit, but this is what happened.  "None of these" is not what I submitted, and is in fact incorrect.  
<img width="466" alt="preview5 5changinganswer" src="https://github.com/RunestoneInteractive/rs/assets/7978733/d60c036f-db5b-4226-ab59-b5a78083703f">

**Comment by Alex-Jordan** [comment]
1.1 does not stick on loading for me. Can you confirm that you are working
with cleared, freshly renewed js files?

On Sat, Oct 28, 2023 at 1:40 PM Tanaquil18 ***@***.***> wrote:

> Preview 5.5
> <https://protect2.fireeye.com/v1/url?k=31323334-501cfaeb-3132feb7-454455535732-cdc97d82d2d2db0f&q=1&e=99b4ba55-95fa-4c4c-947e-df1f8854e2c7&u=https%3A%2F%2Frunestone.academy%2Fns%2Fbooks%2Fpublished%2Ffranciscan_mth162_F23%2Fsec-5-5-other-opt.html%23ACS__Preview__5-5>
> is not restoring answers to the dropdowns, but that was no different than
> before. However, a new behavior which is concerning is that it is changing
> the first answer to "None of these" after hitting submit.
>
> I entered u-sub for the first answer, and hit submit, but this is what
> happened. "None of these" is not what I submitted, and is in fact incorrect.
> [image: preview5 5changinganswer]
> <https://user-images.githubusercontent.com/7978733/278837162-d60c036f-db5b-4226-ab59-b5a78083703f.PNG>
>
> —
> Reply to this email directly, view it on GitHub
> <https://protect2.fireeye.com/v1/url?k=31323334-501cfaeb-3132feb7-454455535732-1e9c0116df25bfb4&q=1&e=99b4ba55-95fa-4c4c-947e-df1f8854e2c7&u=https%3A%2F%2Fgithub.com%2FRunestoneInteractive%2Frs%2Fissues%2F373%23issuecomment-1783916857>,
> or unsubscribe
> <https://protect2.fireeye.com/v1/url?k=31323334-501cfaeb-3132feb7-454455535732-161567b1263feb4e&q=1&e=99b4ba55-95fa-4c4c-947e-df1f8854e2c7&u=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABEDOAGK2DCGUUHQ3X5RPPLYBVUULAVCNFSM6AAAAAA5EHZTEGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTOOBTHEYTMOBVG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Alex Jordan
Mathematics Instructor
Portland Community College

**Comment by Alex-Jordan** [comment]
I do see a new issue on my end unrelated to anything mentioned before. I'm
seeing it restore answers from the previous session, even if in the current
session you just submitted something different. I can probably fix that
quickly by testing if it was the initial load or not.

On Sat, Oct 28, 2023 at 1:52 PM Alex Jordan ***@***.***> wrote:

> 1.1 does not stick on loading for me. Can you confirm that you are working
> with cleared, freshly renewed js files?
>
> On Sat, Oct 28, 2023 at 1:40 PM Tanaquil18 ***@***.***>
> wrote:
>
>> Preview 5.5
>> <https://protect2.fireeye.com/v1/url?k=31323334-501cfaeb-3132feb7-454455535732-cdc97d82d2d2db0f&q=1&e=99b4ba55-95fa-4c4c-947e-df1f8854e2c7&u=https%3A%2F%2Frunestone.academy%2Fns%2Fbooks%2Fpublished%2Ffranciscan_mth162_F23%2Fsec-5-5-other-opt.html%23ACS__Preview__5-5>
>> is not restoring answers to the dropdowns, but that was no different than
>> before. However, a new behavior which is concerning is that it is changing
>> the first answer to "None of these" after hitting submit.
>>
>> I entered u-sub for the first answer, and hit submit, but this is what
>> happened. "None of these" is not what I submitted, and is in fact incorrect.
>> [image: preview5 5changinganswer]
>> <https://user-images.githubusercontent.com/7978733/278837162-d60c036f-db5b-4226-ab59-b5a78083703f.PNG>
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://protect2.fireeye.com/v1/url?k=31323334-501cfaeb-3132feb7-454455535732-1e9c0116df25bfb4&q=1&e=99b4ba55-95fa-4c4c-947e-df1f8854e2c7&u=https%3A%2F%2Fgithub.com%2FRunestoneInteractive%2Frs%2Fissues%2F373%23issuecomment-1783916857>,
>> or unsubscribe
>> <https://protect2.fireeye.com/v1/url?k=31323334-501cfaeb-3132feb7-454455535732-161567b1263feb4e&q=1&e=99b4ba55-95fa-4c4c-947e-df1f8854e2c7&u=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABEDOAGK2DCGUUHQ3X5RPPLYBVUULAVCNFSM6AAAAAA5EHZTEGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTOOBTHEYTMOBVG4>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
>
> --
> Alex Jordan
> Mathematics Instructor
> Portland Community College
>


-- 
Alex Jordan
Mathematics Instructor
Portland Community College

**Comment by Alex-Jordan** [comment]
I went through these three exercises again and they are all working for me.
Except I noticed what I mentioned, about how if I would change my answers
and resubmit, the routine was still repopulating with the selections from a
previous session. But I fixed that and tested on each of these three
exercises.

Thanks for checking things. I think after today I will be obliged not to
tweak anything. So whatever we come up with today will be in place for a
while. For now all I can guess is that one or both of the js files involved
(one on the RS side, one on the PTX side) were cached and not working with
each other. But if you are certain you have non-cached js in play, let me
know. We might need to video conference to work it out live.

On Sat, Oct 28, 2023 at 1:55 PM Alex Jordan ***@***.***> wrote:

> I do see a new issue on my end unrelated to anything mentioned before. I'm
> seeing it restore answers from the previous session, even if in the current
> session you just submitted something different. I can probably fix that
> quickly by testing if it was the initial load or not.
>
> On Sat, Oct 28, 2023 at 1:52 PM Alex Jordan ***@***.***> wrote:
>
>> 1.1 does not stick on loading for me. Can you confirm that you are
>> working with cleared, freshly renewed js files?
>>
>> On Sat, Oct 28, 2023 at 1:40 PM Tanaquil18 ***@***.***>
>> wrote:
>>
>>> Preview 5.5
>>> <https://protect2.fireeye.com/v1/url?k=31323334-501cfaeb-3132feb7-454455535732-cdc97d82d2d2db0f&q=1&e=99b4ba55-95fa-4c4c-947e-df1f8854e2c7&u=https%3A%2F%2Frunestone.academy%2Fns%2Fbooks%2Fpublished%2Ffranciscan_mth162_F23%2Fsec-5-5-other-opt.html%23ACS__Preview__5-5>
>>> is not restoring answers to the dropdowns, but that was no different than
>>> before. However, a new behavior which is concerning is that it is changing
>>> the first answer to "None of these" after hitting submit.
>>>
>>> I entered u-sub for the first answer, and hit submit, but this is what
>>> happened. "None of these" is not what I submitted, and is in fact incorrect.
>>> [image: preview5 5changinganswer]
>>> <https://user-images.githubusercontent.com/7978733/278837162-d60c036f-db5b-4226-ab59-b5a78083703f.PNG>
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://protect2.fireeye.com/v1/url?k=31323334-501cfaeb-3132feb7-454455535732-1e9c0116df25bfb4&q=1&e=99b4ba55-95fa-4c4c-947e-df1f8854e2c7&u=https%3A%2F%2Fgithub.com%2FRunestoneInteractive%2Frs%2Fissues%2F373%23issuecomment-1783916857>,
>>> or unsubscribe
>>> <https://protect2.fireeye.com/v1/url?k=31323334-501cfaeb-3132feb7-454455535732-161567b1263feb4e&q=1&e=99b4ba55-95fa-4c4c-947e-df1f8854e2c7&u=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABEDOAGK2DCGUUHQ3X5RPPLYBVUULAVCNFSM6AAAAAA5EHZTEGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTOOBTHEYTMOBVG4>
>>> .
>>> You are receiving this because you were mentioned.Message ID:
>>> ***@***.***>
>>>
>>
>>
>> --
>> Alex Jordan
>> Mathematics Instructor
>> Portland Community College
>>
>
>
> --
> Alex Jordan
> Mathematics Instructor
> Portland Community College
>


-- 
Alex Jordan
Mathematics Instructor
Portland Community College

**Comment by bnmnetp** [comment]
Chrissy, I don't think it will correctly restore old answers, only questions that you have answered using the build I did yesterday or later.

**Comment by Tanaquil18** [comment]
I think it was cached.  I did a bunch of problems with the console open and caching disabled, and it did not get stuck on loading, multiple choice and dropdowns appeared to all get restored, and I watched in real time as the "changing first multiple choice or dropdown submitted answer to a previous stored answer" problem went away in Preview 5.5 as well as Preview 1.5. 

Brad, it actually did correctly restore old multiple choice or dropdown answers to at least 1 problem, Preview 8.1, that was hardcoded in the source by Alex.  Not sure I noticed it for any other problems though. 

So overall, I think things are working, and are much improved from where we were!

FYI, .pg problems that use the older checkbox macro are still problematic.  I've edited and replaced the vast majority of these, but there are still a few Matt and I have noted and marked for me to edit over the semester or summer break, and those ones are still not restoring all answers.  They also don't tell students anything about right/wrong after hitting "submit".

Thanks to both of you!

**Comment by Alex-Jordan** [comment]
Thanks for all the testing!

> FYI, .pg problems that use the older checkbox macro are still problematic. I've edited and replaced the vast majority of these, but there are still a few Matt and I have noted and marked for me to edit over the semester or summer break, and those ones are still not restoring all answers. They also don't tell students anything about right/wrong after hitting "submit".

Can you give me a path to a problem like this, if it's in the OPL? I will put it in the testing document and see what it does differently that I might be able to account for.

**Comment by Tanaquil18** [comment]
`<webwork source="Library/Westmont/ActiveCalculus/Preview_7_1/preview_7_1_all.pg" />`

but there are others too.  

And this is almost certainly a different issue, but if you're adding testing problems, then in this one, the first dropdown after the redacted essay box has no possible correct answer, even though it works correctly inside of webwork proper.  I've edited the problem that appears in Runestone to take out the essay box entirely, and it works after doing that.  

`<webwork source="Library/Westmont/ActiveCalculus/Preview_6_3/preview_6_3_a.pg" />`

**Comment by Tanaquil18** [comment]
Also, I did just check one that I'd already edited to use parserCheckboxList instead, and that works well, gives feedback upon submission, and restores answers.

**Comment by Alex-Jordan** [comment]
OK, I looked at that first problem and I think I see enough to be able to
say we can't support how it works. It uses PGchoicemacros.pl tools, which
are deprecated. There's just not enough reliable structure between the HTML
input value for the options and the format that the official student answer
has.

What can always be done is updating an OPL problem to use the newer, better
tools. Which it sounds like you are doing. If you are not already doing so,
I would encourage you to edit the actual OPL problem and open a PR with the
edits, and not just create local copies of the files.

On Sat, Oct 28, 2023 at 3:01 PM Tanaquil18 ***@***.***> wrote:

> <webwork source="Library/Westmont/ActiveCalculus/Preview_7_1/
> preview_7_1_all.pg" />
>
> but there are others too.
>
> And this is almost certainly a different issue, but if you're adding
> testing problems, then in this one, the first dropdown after the redacted
> essay box has no possible correct answer, even though it works correctly
> inside of webwork proper. I've edited the problem that appears in Runestone
> to take out the essay box entirely, and it works after doing that.
>
> <webwork source="Library/Westmont/ActiveCalculus/Preview_6_3/
> preview_6_3_a.pg" />
>
> —
> Reply to this email directly, view it on GitHub
> <https://protect2.fireeye.com/v1/url?k=31323334-501d2dca-3132feb7-454455534531-bda2b5534ee12a30&q=1&e=fcaef838-7889-4cf4-b0df-6933dd66f3fd&u=https%3A%2F%2Fgithub.com%2FRunestoneInteractive%2Frs%2Fissues%2F373%23issuecomment-1783929555>,
> or unsubscribe
> <https://protect2.fireeye.com/v1/url?k=31323334-501d2dca-3132feb7-454455534531-fdac2a20cd53c9af&q=1&e=fcaef838-7889-4cf4-b0df-6933dd66f3fd&u=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABEDOAHELIBW4CO7YJL42GDYBV6ENAVCNFSM6AAAAAA5EHZTEGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTOOBTHEZDSNJVGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Alex Jordan
Mathematics Instructor
Portland Community College

**Comment by Tanaquil18** [comment]
Makes sense.  I already updated most of the problems that appear in AC anyway, with just 4 or 5 more that we've noticed so far.

I am mostly just making local edits.  I did try to submit 2 edits back to the OPL in July, and the pull requests are still open.  That isn't very motivating for me to spend time doing more of that.  Is there some comment that I'm not seeing and addressing, or do you know what the holdup is?

One of them, which included changing checkbox macros, was simple enough that I swapped to PGML also. [link to pr on github](https://github.com/openwebwork/webwork-open-problem-library/pull/1069)

The other problem was more complicated, and I'm much more experienced using PG than I am with PGML, and so I only updated the table macro. I don't know if there's something else I'm missing, or if it's pointless to try submitting edits back if I can't spend the time to make the problems fully new.   [Link to pr on Github](https://github.com/openwebwork/webwork-open-problem-library/pull/1071)

**Comment by bnmnetp** [comment]
Closing this one out. @Alex-Jordan's work on restoring radio-button and dropdown answers landed, and @Tanaquil18 confirmed on 2023-10-28 that the Active Calculus examples were restoring correctly ("overall, I think things are working, and are much improved from where we were").

The cases still failing at that point were `.pg` problems built on the deprecated `PGchoicemacros.pl` tools, where there isn't enough reliable structure between the HTML input values and the stored student answer — the fix for those is updating the OPL problem, not the Runestone side. Ongoing WeBWorK version-compatibility work is tracked in #319.

If a specific problem still doesn't restore answers, a new issue naming the `.pg` path will be much easier to chase than this thread.
